services/workflows-service/src/alert/alert.service.ts

```diff
@@ -140,15 +151,66 @@ export class AlertService {
     }
   }
 
-  // Specific alert check logic based on the definition
-  private async checkAlert(alertDefinition: AlertDefinition) {
+  async checkOngoingMonitoringAlert(
```

Comment:
can you use same check alert function and merge it as a factory once the alert type is related to ongoing merchant report?

this function can replace by factory the create alert args and continue to check if we should set a dedupe mechanisem once we already have an open alert

talk to me about it

Reply:
@nitzanballerine